core-it-suite/src/test/java/org/apache/maven/it/MavenIT0146InstallerSnapshotNaming.java

```diff
@@ -59,15 +60,11 @@ protected void setUp()
         server = new Server( 0 );
         server.setHandler( handlers );
         server.start();
-        while ( !server.isRunning() || !server.isStarted() )
+        if ( server.isFailed() )
         {
-            if ( server.isFailed() )
-            {
-                fail( "Couldn't bind the server socket to a free port!" );
-            }
-            Thread.sleep( 100L );
+            fail( "Couldn't bind the server socket to a free port!" );
         }
-        port = server.getConnectors()[0].getLocalPort();
+        port = ( (NetworkConnector) server.getConnectors()[0] ).getLocalPort();
```

Comment:
There's also `server.getURI().getPort()` as a valid option here.

Reply:
I did not want to make the code too much different from what we had before.I was glad that the version 9.0.4 works. After I changed the version to 9.2.28 I got HTTP 405 and really did not want to get new arguments from the devs saying that there is something which broke new version.

Reply:
We can improve as always but I spent 3 days with these 38 tests and for me it is always good to have several smaller changes than one big change. With small changes/commits we can always find test older commits if we find some issue.

Reply:
So I would make a new commit as an improvement of this line.

Reply:
@joakime 
Do you see more occurences of this? Notice that we also have two connectors in few tests using HTTPS.